src/main/kotlin/kweb/util/misc.kt

```diff
@@ -30,6 +36,32 @@ fun String.escapeEcma() = StringEscapeUtils.escapeEcmaScript(this)!!
 
 fun Any.toJson(): String = gson.toJson(this)
 
+fun primitiveToJson(value: Any?, errorMsg: String = "Argument is required to be String or primitive type"): JsonElement {
```

Comment:
Why is errorMsg passed in as a parameter?

Reply:
Because this function is used in a few different places. Passing in an errorMsg lets the function that calls it specify where it's happening. It was the best way to have robust error messages, while avoiding code duplication.